setup.py

```diff
@@ -19,11 +19,11 @@
 
 
 def version():
-    with open("mlrun/__init__.py") as fp:
-        for line in fp:
-            if "__version__" in line:
-                _, version = line.split("=")
-                return version.replace('"', "").strip()
+    with open("version.json") as version_file:
+        for line in version_file:
+            if '"version"' in line:
+                # '  "version": "<version>",' -> ['  ', 'version', ': ', '<version>', ',']
+                return line.split('"')[3]
```

Comment:
I think doing this here will be safer code:
```
with open("version.json") as version_file:
    try:
        version_metadata = json.load(version_file)
        return version_metadata['version']
    except (ValueError, KeyError):
        return None

Reply:
This is json. parse it as such
Also if possible (without circular import issues) to bring in a logger here and output something in case of error inferring the version - it might be very useful, otherwise missing version information will be hard to debug

Reply:
tl;dr - `json` yes, logger no
I was thinking that it is impossible to import stuff in `setup.py` cause it's running before installation (before installing the requirements) but apparently it is possible using the `pyproject.toml` file (https://www.python.org/dev/peps/pep-0518/#build-system-table), also I wasn't aware but apparently anyway `json` is part of the standard library so need to pre-install anything. w.r.t. logger, it is impossible, I can't use the package here, because it is not installed yet

Reply:
logger is python standard library :)

Reply:
Indeed, thought you meant MLRun logger, added